Fixes for app build process

sebworks opened the pull request:
Fixes for app build process

## Changes

- Modified `scripts/npm/apps-install/index.js` to use `execSync` process to avoid race conditions with `gulp build`.  Added `process.exit` to stop build process on failure. Included apps package.*json contents in checksum frontend.sh process.

## Testing

1. Run `frontend.sh production` and verify the console content is as expected for the apps.
2. Modify the owning-a-home package.json to bump the jquery version to 5
3. Run `frontend.sh production` again, you should see an error in the console and the build should stop.


## Screenshots


## Notes

-

## Todos

-

## Checklist

* [ ] PR has an informative and human-readable title
* [ ] Changes are limited to a single goal (no scope creep)
* [ ] Code can be automatically merged (no conflicts)
* [ ] Code follows the standards laid out in the [development playbook](https://github.com/cfpb/development)
* [ ] Passes all existing automated tests
* [ ] Any *change* in functionality is tested
* [ ] New functions are documented (with a description, list of inputs, and expected output)
* [ ] Placeholder code is flagged
* [ ] Visually tested in supported browsers and devices
* [ ] Project documentation has been updated
* [ ] Reviewers requested with the [Reviewer tool](https://help.github.com/articles/about-pull-request-reviews/) :arrow_right:

anselmbradford commented:
> Run frontend.sh production again, you should see an error in the console and the build should stop.

Hmm this didn't happen for me. I updated `unprocessed/apps/owning-a-home/package.json` to have `"jquery": "^5.0.0",`

sebworks commented:
closing in favor of #3834

sebworks commented:
>> Hmm this didn't happen for me. I updated unprocessed/apps/owning-a-home/package.json to have "jquery": "^5.0.0",


The reason it wasn't running was because we run `npm install --production` in `frontend.sh`, which doesn't install devDependencies. I've moved them to dependencies in https://github.com/cfpb/cfgov-refresh/pull/3840/commits/0864fa8a8c13319a45e43aaf41027fee3c46d3c1.

anselmbradford commented on frontend.sh:
```diff
@@ -15,9 +15,9 @@ init() {
 
   if [ -f "package-lock.json" ]; then
-    DEP_CHECKSUM=$(cat package-lock.json package.json | shasum -a 256)
+    DEP_CHECKSUM=$(cat package-lock.json package.json cfgov/unprocessed/apps/**/package*.json | shasum -a 256)
```
I supposed based on your edit this could  be `DEP_CHECKSUM=$(cat package*.json cfgov/unprocessed/apps/**/package*.json | shasum -a 256)`

sebworks replied:
good point, fixed in https://github.com/cfpb/cfgov-refresh/pull/3840/commits/c5a8c4cc880c5de2ac66b33972d0515de7fa74ff

anselmbradford commented on frontend.sh:
```diff
@@ -15,9 +15,9 @@ init() {
+    DEP_CHECKSUM=$(cat package-lock.json package.json cfgov/unprocessed/apps/**/package*.json | shasum -a 256)
   else
-    DEP_CHECKSUM=$(cat package.json | shasum -a 256)
+    DEP_CHECKSUM=$(cat package.json cfgov/unprocessed/apps/**/package*.json | shasum -a 256)
```
Should this include `package*.json` if `package-lock.json` isn't included in the main one?

sebworks replied:
fixed in https://github.com/cfpb/cfgov-refresh/pull/3840/commits/c5a8c4cc880c5de2ac66b33972d0515de7fa74ff
